src/sema/expression.rs

```diff
@@ -6218,7 +6229,7 @@ fn resolve_using(
                     }
                 };
 
-                match cast(&arg.loc(), arg.clone(), &ty, true, ns, &mut errors) {
+                match arg.cast(&arg.loc(), &ty, true, ns) {
                     Ok(expr) => cast_args.push(expr),
```

Comment:
again here the cast error should go to `errors`, not `ns.diagnostics`. If the argument cannot be casted, then try the next library function with same name.